apps/projects/routes.py

```diff
@@ -176,100 +377,814 @@ def get(self, project_id):
+        if not doc['metadata']:
+            return not_found('Video is still processing')
+
+        # get thumbnails of video
+        if request.args.get('thumbnail'):
+            thumbnail = request.args.get('thumbnail')
+            if thumbnail == 'preview':
+                preview_thumbnail = doc.get('preview_thumbnail')
+                if not preview_thumbnail:
+                    return not_found('')
+                byte = app.fs.get(preview_thumbnail.get('storage_id'))
+                res = make_response(byte)
+                res.headers['Content-Type'] = 'image/png'
+
+            else:
+                thumbnail = represents_int(thumbnail)
+                if thumbnail or thumbnail == 0:
+                    total = list(doc['thumbnails'].keys())[0]
+                    if not thumbnail >= 0 or thumbnail >= len(doc['thumbnails'][total]):
+                        return not_found('')
+                    byte = app.fs.get(doc['thumbnails'][total][thumbnail]['storage_id'])
+                    res = make_response(byte)
+                    res.headers['Content-Type'] = 'image/png'
+                else:
+                    res = bad_request("thumbnail variable must be int or 'preview'")
+            return res
+
+        # get strem file for video
+        video_range = request.headers.environ.get('HTTP_RANGE')
+        length = int(doc['metadata'].get('size'))
+        if video_range:
+
+            start = int(re.split('[= | -]', video_range)[1])
+            end = length - 1
+            chunksize = end - start + 1
+            headers = {
+                'Content-Range': f'bytes {start}-{end}/{length}',
+                'Accept-Ranges': 'bytes',
+                'Content-Length': chunksize,
+                'Content-Type': doc.get("mime_type"),
+            }
+            # get a stack of bytes push to client
+            stream = app.fs.get_range(doc['storage_id'], start, chunksize)
+            res = make_response(stream)
+            res.headers = headers
+            return res, 206
+        else:
+
+            headers = {
+                'Content-Length': length,
+                'Content-Type': 'video/mp4',
+            }
+            stream = app.fs.get(doc.get('storage_id'))
+            res = make_response(stream)
+            res.headers = headers
+            return res, 200
+
+
+class PreviewThumbnailVideo(MethodView):
+    SCHEMA_PREVIEW_THUMBNAIL = {
+        'type': {
+            'type': 'string',
+            'required': True,
+            'empty': False,
+            'allowed': ['capture', 'upload'],
+        },
+        'time': {
+            'type': 'float',
+            'required': False,
+            'empty': True,
+            'nullable': True,
+            'min': 0,
+        },
+        'data': {
+            'type': 'string',
+            'required': False,
+            'empty': True,
+            'nullable': True,
+        },
+    }
+
+    def post(self, project_id):
```

Comment:
I would  use `post` for upload thumbnail and `get` for capture. We use `get` in `ThumbnailsTimelineProject` for almost the same action. does it make sense?
P.S.
Does it makes sense to create only one generic view and call it let's say `RetrieveOrCreateThumbnails`, so it will handle all thumbnails creation?

Reply:
Yes, I also want to create only one for all thumbnails. 
But how we can distinct the get preview thumbnail and time-line thumbnails via this designed url `projects/<path:project_id>/thumbnails`.
Maybe get preview thumbnail we haven't amount in query `projects/<path:project_id>/thumbnails`.
And get time-line thumbnails  is `projects/<path:project_id>/thumbnails?amount=<number>`.

Reply:
I think we can use something like this for timeline thumbnails:
`projects/<path:project_id>/thumbnails?type=timeline&amount=10`
and for preview:
`projects/<path:project_id>/thumbnails?type=preview&time=5`